analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/LoopDetector.cs

```diff
@@ -83,13 +83,22 @@ void MergeWithIntersectingLoops(int[] path, int last)
         }
     }
 
-    private static IEnumerable<int> Successors(ControlFlowGraph cfg, int block)
+    private static IEnumerable<int> Successors(ControlFlowGraph cfg, int index)
     {
-        var successors = cfg.Blocks[block].SuccessorBlocks.Select(x => x.Ordinal);
-        if (cfg.Blocks[block].EnclosingRegion(ControlFlowRegionKind.Try) is { } tryRegion)
+        var block = cfg.Blocks[index];
+        var successors = block.SuccessorBlocks.Select(x => x.Ordinal);
+        if (block.EnclosingRegion(ControlFlowRegionKind.Try) is { } tryRegion)
         {
             successors = successors.Concat(tryRegion.ReachableHandlers().Select(x => x.FirstBlockOrdinal));
         }
+        if (block.EnclosingRegion(ControlFlowRegionKind.Finally) is { } finallyRegion)
+        {
+            var tryFinally = finallyRegion.EnclosingRegion;
```

Comment:
Just ask for it directly
```suggestion
        if (block.EnclosingRegion(ControlFlowRegionKind.TryFinally) is { } tryFinally)
        {
```

Reply:
This changes the logic, doesn't it? With your change, this would be reachable from a `try` region.

Reply:
Indeed